src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -5227,6 +5108,35 @@ static void Main()
         }
 
         [Fact]
+        public void VerifyDiscardIL()
+        {
+            var source =
+@"
+class C
+{
+    static int M()
+    {
+        var (x, _, _) = (1, new C(), 2);
+        return x;
+    }
+}
+";
+
+            var comp = CompileAndVerify(source, additionalRefs: s_valueTupleRefs);
+            comp.VerifyDiagnostics();
+            comp.VerifyIL("C.M()", @"
+{
+  // Code size        8 (0x8)
+  .maxstack  1
+  .locals init (C V_0)
+  IL_0000:  newobj     ""C..ctor()""
+  IL_0005:  stloc.0
```

Comment:
You have a pattern : ` unused1 = unused2 = Expr`
Optimizer can see that `unused1` is never read from, but it is not as smart to figure that `unused2` is also unused. 

That should be ok for the purpose of this change. 

Many discards should not even make it to the codegen, but, IMO, optimizing discards specifically should be done separately.

---
In reply to: [100868940](https://github.com/dotnet/roslyn/pull/17027#discussion_r100868940) [](ancestors = 100868940)